src/lib/types/EmbeddingEndpoints.ts

```diff
@@ -0,0 +1,39 @@
+import { z } from "zod";
+import {
+	embeddingEndpointTei,
+	embeddingEndpointTeiParametersSchema,
+} from "$lib/server/embeddingEndpoints/tei/embeddingEndpoints";
+import {
+	embeddingEndpointTransformersJS,
+	embeddingEndpointTransformersJSParametersSchema,
+} from "$lib/server/embeddingEndpoints/transformersjs/embeddingEndpoints";
+
+// parameters passed when generating text
+interface EmbeddingEndpointParameters {
+	inputs: string[];
+}
+
+// type signature for the endpoint
+export type EmbeddingEndpoint = (params: EmbeddingEndpointParameters) => Promise<number[][]>;
```

Comment:
You can extract `number[][]` into type, such as `type Embedding = number[][]` or something like that